op-node/rollup/derive/channel_out_test.go

```diff
@@ -184,3 +219,94 @@ func TestBlockToBatchValidity(t *testing.T) {
+	// target is larger than one batch, but smaller than two batches
+	target := uint64(300)
+	rng := rand.New(rand.NewSource(0x543331))
+	chainID := big.NewInt(rng.Int63n(1000))
+	txCount := 1
+	cout, err := NewSpanChannelOut(0, chainID, target)
+	require.NoError(t, err)
+
+	compressedLens := []int{}
+
+	// adding the first batch should not cause an error
+	singularBatch := RandomSingularBatch(rng, txCount, chainID)
+	_, err = cout.AddSingularBatch(singularBatch, 0)
+	compressedLens = append(compressedLens, cout.compressed.Len())
+	require.NoError(t, err)
+
+	// confirm compression was not skipped
+	require.Greater(t, compressedLens[0], 0)
+	// confirm the channel is full
+	require.ErrorIs(t, cout.FullErr(), ErrCompressorFull)
+}
+
+// TestSpanChannelOutCompressionUndo tests that the SpanChannelOut compression works as expected to not accept a second batch
+// if that batch would cause the channel to exceed the target size. The first batch should be compressed only
+// the channel should be compressed, full, and should not have the second batch
+func TestSpanChannelOutCompressionUndo(t *testing.T) {
+	// target is larger than one batch, but smaller than two batches
+	target := uint64(750)
+	rng := rand.New(rand.NewSource(0x543331))
+	chainID := big.NewInt(rng.Int63n(1000))
+	txCount := 1
+	cout, err := NewSpanChannelOut(0, chainID, target)
+	require.NoError(t, err)
+
+	compressedLens := []int{}
+	rlpLens := []int{}
+
+	// adding the first batch should not cause an error
+	singularBatch := RandomSingularBatch(rng, txCount, chainID)
+	_, err = cout.AddSingularBatch(singularBatch, 0)
+	compressedLens = append(compressedLens, cout.compressed.Len())
+	rlpLens = append(rlpLens, cout.activeRLP().Len())
+	require.NoError(t, err)
+
+	// adding the second batch should cause an error
+	singularBatch = RandomSingularBatch(rng, txCount, chainID)
+	_, err = cout.AddSingularBatch(singularBatch, 0)
+	compressedLens = append(compressedLens, cout.compressed.Len())
+	rlpLens = append(rlpLens, cout.activeRLP().Len())
+	require.ErrorIs(t, err, ErrCompressorFull)
+
+	// confirm that the first compression was skipped
+	require.Equal(t, 0, compressedLens[0])
+	// confirm that the second compression was not skipped
+	require.Greater(t, compressedLens[1], 0)
+	// confirm that the second rlp is tht same size as the first (because the second batch was not added)
+	require.Equal(t, rlpLens[0], rlpLens[1])
+}
+
+// TestSpanChannelOutClose tests that the SpanChannelOut compression works as expected when the channel is closed.
+// it should compress the batch even if it is smaller than the target size because the channel is closing
+func TestSpanChannelOutClose(t *testing.T) {
+	// target is larger than one batch
+	target := uint64(600)
+	rng := rand.New(rand.NewSource(0x543331))
+	chainID := big.NewInt(rng.Int63n(1000))
+	txCount := 1
+	cout, err := NewSpanChannelOut(0, chainID, target)
+	require.NoError(t, err)
+
+	singularBatch := RandomSingularBatch(rng, txCount, chainID)
```

Comment:
the whole setup code of the `cout` and `singularBatch` could be put into a separate setup function `func setupSpanChannelOutTest(t *testing.T, target uint64) (*SpanChannelOut, *SingularBatch)` and deduplicate in a lot of tests, including the 3 new tests.